tests/phpunit/integration/Core/Authentication/Google_ProxyTest.php

```diff
@@ -43,6 +44,134 @@ public function test_get_site_fields() {
+			'return_uri',
+			'action_uri',
+			'analytics_redirect_uri',
+		);
+		$this->mock_http_request(
+			$google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ),
+			$mock_response
+		);
+		$google_proxy->fetch_site_fields( $credentials );
+
+		// Ensure the request was made with the proper URL and body parameters.
+		$this->assertEquals( $google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ), $pre_url );
+		$this->assertEquals( 'POST', $pre_args['method'] );
+
+		$this->assertEqualSets(
+			array(
+				'site_id',
+				'site_secret',
+			),
+			array_keys( $pre_args['body'] )
+		);
+	}
+
+	public function test_are_site_fields_synced() {
+		$context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$google_proxy = new Google_Proxy( $context );
+		$credentials  = new Credentials( new Options( $context ) );
+
+		$fake_creds = $this->fake_proxy_site_connection();
+
+		$pre_args = null;
+		$pre_url  = null;
+
+		add_filter(
+			'pre_http_request',
+			function ( $false, $args, $url ) use ( &$pre_args, &$pre_url ) {
+				$pre_args = $args;
+				$pre_url  = $url;
+
+				return $false;
+			},
+			10,
+			3
+		);
+
+		// Mock matching reponse.
+		$matching_mock_response = array(
+			'url'                    => home_url(),
+			'action_uri'             => admin_url( 'index.php' ),
+			'name'                   => get_bloginfo( 'name' ),
+			'return_uri'             => $context->admin_url( 'splash' ),
+			'redirect_uri'           => add_query_arg( 'oauth2callback', 1, admin_url( 'index.php' ) ),
+			'analytics_redirect_uri' => add_query_arg( 'gatoscallback', 1, admin_url( 'index.php' ) ),
+		);
+		$this->mock_http_request(
+			$google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ),
+			$matching_mock_response
+		);
+		$success_response_data = $google_proxy->are_site_fields_synced( $credentials );
+
+		// Ensure the request was made with the proper URL and body parameters.
+		$this->assertEquals( $google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ), $pre_url );
+		$this->assertEquals( 'POST', $pre_args['method'] );
+		$this->assertEqualSetsWithIndex(
+			array(
+				'site_id',
+				'site_secret',
+			),
+			array_keys( $pre_args['body'] )
+		);
+
+		// Mock WP_Error response
+		$mock_wp_error_response = new WP_Error();
+		$this->mock_http_request(
+			$google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ),
+			$mock_wp_error_response
+		);
+		$error_response_data = $google_proxy->are_site_fields_synced( $credentials );
+
+		// Mock non matching response.
+		$mock_non_matching_response = array(
+			'incorrect',
+			'keys',
+		);
+		$this->mock_http_request(
+			$google_proxy->url( Google_Proxy::OAUTH2_SITE_URI ),
+			$mock_non_matching_response
+		);
+		$failure_response_data = $google_proxy->are_site_fields_synced( $credentials );
+
+		// Ensure matching response array returns true.
+		$this->assertEquals( $success_response_data, true );
+		// Ensure WP_Error response returns WP_Error.
+		$this->assertEquals( is_wp_error( $error_response_data ), true );
```

Comment:
@benbowler let's use `$this->assertWPErrorWithMessage`, defined in the `tests/phpunit/includes/TestCase.php` file, to better evaluate the error object.